src/Form/CitizenProjectCommandType.php

```diff
@@ -143,5 +143,16 @@ public function preSetData(FormEvent $event)
                 'disabled' => true,
             ]);
         }
+
+        if (!$citizenProject instanceof CitizenProject) {
```

Comment:
What do you think if we change this condition to 
```
if (!$citizenProject) {
```

or 

```
if (null === $citizenProject) {
```

Reply:
@HeahDude prefere instanceof... but for me I don't have preference...

Reply:
Ok as you wish guys, not a big deal.
I just thought it would be little bit more efficient like this.

Reply:
>@HeahDude prefere instanceof... 

Not in all cases...

In fact I don't get why there is a pre set data listener, the condition is already there in `buildForm`. Please merge the logic in one place by removing this listener. 

The call here https://github.com/EnMarche/en-marche.fr/pull/1856/files#diff-76eb4209f0717100613e396a1c32ebc2R95 might break since `$builder->getData()` is no tested. Please add before something like:
```php
$command = $builder->getData();

if (!$commmand instanceof CitizenProjectCommand) {
    throw new InvalidConfigurationException('A pre set data is required in '.__CLASS__);
}
```